tests/Integration/Infrastructure/Persistence/MySQLAssetRepositoryTest.php

```
@@ -0,0 +1,784 @@
+     * @param array{host: string, port: int, user: string, password: string} $connection
+     */
+    private function databaseDsn(array $connection, string $databaseName): string
+    {
+        return sprintf(
+            'mysql:host=%s;port=%d;dbname=%s;charset=utf8mb4',
+            $connection['host'],
+            $connection['port'],
+            $databaseName,
+        );
+    }
+}
```

PHPMD (2.15.0)

[warning] The class MySQLAssetRepositoryTest has 31 non-getter- and setter-methods. Consider refactoring MySQLAssetRepositoryTest to keep number of methods under 25. (undefined)

(TooManyMethods)